apps/prairielearn/src/ee/lib/validateHTML.ts

```diff
@@ -584,7 +586,15 @@ function dfsCheckParseTree(ast: DocumentFragment | ChildNode, optimistic: boolea
  * @param optimistic True if tags outside the subset are allowed, else false.
  * @returns A list of human-readable render error messages, if any.
  */
-export function validateHTML(file: string, optimistic: boolean): string[] {
+export function validateHTML(file: string, optimistic: boolean, usesServerPy: boolean): string[] {
   const tree = parse5.parseFragment(file);
+  const templates = [...file.matchAll(mustacheTemplateExtractorRegex)]
+    .map((x) => x[1])
+    .concat([...file.matchAll(answersNameExtractorRegex)].map((x) => `correct_answers.${x[1]}`));
+  if (!usesServerPy && templates.length > 0) {
+    return dfsCheckParseTree(tree, optimistic).concat([
+      `Create a server.py to generate the following: ${templates.join(', ')}`,
+    ]);
+  }
   return dfsCheckParseTree(tree, optimistic);
```

Comment:
Consider structuring like this instead:

```suggestion
  const errors = dfsCheckParseTree(tree, optimistic);
  
  if (!usesServerPy && templates.length > 0) {
    errors.push(
      `Create a server.py file to generate the following: ${templates.join(', ')}`,
    );
  }
  
  return errors;
```